cpp/src/parquet/reader_test.cc

```diff
@@ -1059,4 +1060,165 @@ TEST(TestFileReader, TestOverflowInt16PageOrdinal) {
   }
 }
 
+struct PageIndexReaderParam {
+  PageIndexReaderParam(const std::vector<int32_t>& row_group_indices,
+                       const std::vector<int32_t>& column_indices, bool need_column_index,
+                       bool need_offset_index)
+      : row_group_indices(row_group_indices),
+        column_indices(column_indices),
+        index_selection{need_column_index, need_offset_index} {}
```

Comment:
I don't think you need to write this constructor explicitly, aggregate initialization will work automatically.